chore: remove global changelog script since lerna fixed version does this for us now

bryceosterhaus opened the pull request:
Partially addresses #3939

Example of new output below

----------
## 2021-2-18

### @clayui/card ([3.7.0](https://github.com/liferay/clay/compare/@clayui/card@3.6.6...@clayui/card@3.7.0))

#### Features

-   add active state for card when it is selected ([607da9d](https://github.com/liferay/clay/commit/607da9d))

### @clayui/color-picker ([3.5.1](https://github.com/liferay/clay/compare/@clayui/color-picker@3.5.0...@clayui/color-picker@3.5.1))

#### Bug Fixes

-   remove last-row classes and API that was added on accident ([d4cb8e8](https://github.com/liferay/clay/commit/d4cb8e8))

### @clayui/css ([3.24.2](https://github.com/liferay/clay/tree/master/packages/clay-css/compare/@clayui/css@3.24.1...@clayui/css@3.24.2))

#### Bug Fixes

-   Global Functions moves generated SVG icons Sass map to separate file `_lx-icons-generated.scss` ([7ffc523](https://github.com/liferay/clay/tree/master/packages/clay-css/commit/7ffc523))

### @clayui/drop-down ([3.10.0](https://github.com/liferay/clay/compare/@clayui/drop-down@3.9.1...@clayui/drop-down@3.10.0))

#### Features

-   add util for aligning dropdown on scroll ([09dc506](https://github.com/liferay/clay/commit/09dc506))

### @clayui/panel ([3.4.0](https://github.com/liferay/clay/compare/@clayui/panel@3.3.6...@clayui/panel@3.4.0))

#### Features

-   add 'show' class to collapsable panel's button ([1e7f64a](https://github.com/liferay/clay/commit/1e7f64a))

### Packages with a version bump only

#### @clayui/autocomplete ([3.2.7](https://github.com/liferay/clay/compare/@clayui/autocomplete@3.2.6...@clayui/autocomplete@3.2.7))

#### @clayui/breadcrumb ([3.3.7](https://github.com/liferay/clay/compare/@clayui/breadcrumb@3.3.6...@clayui/breadcrumb@3.3.7))

#### @clayui/date-picker ([3.8.1](https://github.com/liferay/clay/compare/@clayui/date-picker@3.8.0...@clayui/date-picker@3.8.1))

#### @clayui/list ([3.4.8](https://github.com/liferay/clay/compare/@clayui/list@3.4.7...@clayui/list@3.4.8))

#### @clayui/localized-input ([3.2.7](https://github.com/liferay/clay/compare/@clayui/localized-input@3.2.6...@clayui/localized-input@3.2.7))

#### @clayui/multi-select ([3.9.5](https://github.com/liferay/clay/compare/@clayui/multi-select@3.9.4...@clayui/multi-select@3.9.5))

#### @clayui/multi-step-nav ([3.3.7](https://github.com/liferay/clay/compare/@clayui/multi-step-nav@3.3.6...@clayui/multi-step-nav@3.3.7))

#### @clayui/pagination-bar ([3.2.7](https://github.com/liferay/clay/compare/@clayui/pagination-bar@3.2.6...@clayui/pagination-bar@3.2.7))

#### @clayui/pagination ([3.3.7](https://github.com/liferay/clay/compare/@clayui/pagination@3.3.6...@clayui/pagination@3.3.7))

wincent commented on scripts/generate-global-changelog.js:
```diff
@@ -95,7 +101,9 @@ function main() {
 	const files = diffOutput.split('\0').filter(Boolean);
 
-	let output = `# CHANGELOG\n## ${releaseDateString}\n`;
+	let outputChanges = `# CHANGELOG\n## ${releaseDateString}\n`;
+
+	let versionBumps = `### Packages with a version bump only\n`;
```
Probably never going to happening, but if there are ever any times when we _don't_ have a version bump, this would print an empty heading. Even though it's unlikely, it's good to handle "null state" cases, so that could be something like:

```js
const versionBumps = [];

// ...

if (bumpOnly) {
    versionBumps.push(...);
}

// ...

if (versionBumps.length) {
    outputChanges += '\n### Packages with a version bump only\n' +
        versionBumps.join('\n\n');
}
```

or something similar.

wincent reviewed (approved):
lg2m

wincent commented:
Just a thought that a list might format better:

---

## 2021-2-18

### @clayui/card ([3.7.0](https://github.com/liferay/clay/compare/@clayui/card@3.6.6...@clayui/card@3.7.0))

#### Features

-   add active state for card when it is selected ([607da9d](https://github.com/liferay/clay/commit/607da9d))

### @clayui/color-picker ([3.5.1](https://github.com/liferay/clay/compare/@clayui/color-picker@3.5.0...@clayui/color-picker@3.5.1))

#### Bug Fixes

-   remove last-row classes and API that was added on accident ([d4cb8e8](https://github.com/liferay/clay/commit/d4cb8e8))

### @clayui/css ([3.24.2](https://github.com/liferay/clay/tree/master/packages/clay-css/compare/@clayui/css@3.24.1...@clayui/css@3.24.2))

#### Bug Fixes

-   Global Functions moves generated SVG icons Sass map to separate file `_lx-icons-generated.scss` ([7ffc523](https://github.com/liferay/clay/tree/master/packages/clay-css/commit/7ffc523))

### @clayui/drop-down ([3.10.0](https://github.com/liferay/clay/compare/@clayui/drop-down@3.9.1...@clayui/drop-down@3.10.0))

#### Features

-   add util for aligning dropdown on scroll ([09dc506](https://github.com/liferay/clay/commit/09dc506))

### @clayui/panel ([3.4.0](https://github.com/liferay/clay/compare/@clayui/panel@3.3.6...@clayui/panel@3.4.0))

#### Features

-   add 'show' class to collapsable panel's button ([1e7f64a](https://github.com/liferay/clay/commit/1e7f64a))

### Packages with a version bump only

- @clayui/autocomplete ([3.2.7](https://github.com/liferay/clay/compare/@clayui/autocomplete@3.2.6...@clayui/autocomplete@3.2.7))
- @clayui/breadcrumb ([3.3.7](https://github.com/liferay/clay/compare/@clayui/breadcrumb@3.3.6...@clayui/breadcrumb@3.3.7))
- @clayui/date-picker ([3.8.1](https://github.com/liferay/clay/compare/@clayui/date-picker@3.8.0...@clayui/date-picker@3.8.1))
- @clayui/list ([3.4.8](https://github.com/liferay/clay/compare/@clayui/list@3.4.7...@clayui/list@3.4.8))
- @clayui/localized-input ([3.2.7](https://github.com/liferay/clay/compare/@clayui/localized-input@3.2.6...@clayui/localized-input@3.2.7))
- @clayui/multi-select ([3.9.5](https://github.com/liferay/clay/compare/@clayui/multi-select@3.9.4...@clayui/multi-select@3.9.5))
- @clayui/multi-step-nav ([3.3.7](https://github.com/liferay/clay/compare/@clayui/multi-step-nav@3.3.6...@clayui/multi-step-nav@3.3.7))
- @clayui/pagination-bar ([3.2.7](https://github.com/liferay/clay/compare/@clayui/pagination-bar@3.2.6...@clayui/pagination-bar@3.2.7))
- @clayui/pagination ([3.3.7](https://github.com/liferay/clay/compare/@clayui/pagination@3.3.6...@clayui/pagination@3.3.7))

bryceosterhaus commented:
@wincent since we moved from "independent" to "fixed" modes for lerna, it will automatically generate a global changelog for us. I updated this PR to just remove the script

wincent commented on package.json:
```diff
@@ -34,7 +34,7 @@
 		"storybook": "start-storybook -p 8888",
 		"test": "jest",
-		"version": "yarn genGlobalChangeLog && prettier --write -- \"**/CHANGELOG.md\" && git add -- \"CHANGELOG.md\" \"packages/clay-css/src/scss/_license-text.scss\""
+		"version": "prettier --write -- \"**/CHANGELOG.md\" && git add -- \"CHANGELOG.md\" \"packages/clay-css/src/scss/_license-text.scss\""
```
Out of curiosity, do we still need the `prettier` call here? Before you added the global changelog script this line used to look [like this](https://github.com/liferay/clay/blame/d7e00467ac6e72353ffd69d7841fcfbf4bcfb86a/package.json#L36):

```
		"version": "yarn run format:all"
```

Is the current version better because it's faster?

bryceosterhaus replied:
@wincent yes this version is is more specific for what we actually want and it is faster. The changelog genreation that happens via lerna doesn't follow prettier formatting so we have to make sure and run that before we have the "publish" commit

wincent reviewed (approved):
> I updated this PR to just remove the script

![miss-already](https://user-images.githubusercontent.com/7074/108816383-5fff5a00-75b6-11eb-8364-8930debd5f47.gif)

bryceosterhaus commented:
merging this and gonna work out any kinks during today's release
